### PR TITLE
[7029] Split lib.get_resource_parent

### PR DESCRIPTION
**Pull request by Nishant2022**

### CODE REVIEW

File: scripts/irods/test/test_targeting_specific_replica_number.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def tearDown(self):
         self.user.run_icommand(['irm', '-f', self.logical_path])
 
         for parent, child in [(self.pt1, self.ufs1), (self.pt2, self.ufs2)]:
-            if (lib.get_resource_parent(self.admin, child) == parent):
+            if (lib.get_resource_parent_name(self.admin, child) == parent):
```
[the remark this answers is not in the document]
Comment (Nishant2022) [reply scripts/irods/test/test_targeting_specific_replica_number.py:74]: Yes that is the only place `lib.get_resource_parent` was used.

**Review by korydraughn — COMMENTED** [review]
Code changes look fine.

If you can confirm that they work locally, the I think this PR is good to go.

**Comment by korydraughn** [comment]
Consider looking at #7152 since you're already looking at lib.py.

**Comment by Nishant2022** [comment]
The `test_targeting_specific_replica_number` test which contained the only instance of the `lib.get_resource_parent` function passes with my changes. I will look into the other issue you mentioned.

**Comment by korydraughn** [comment]
Excellent.

We can proceed with getting this merged as-is or you can add additional commits for the other issue.

Do you have a preference?

**Comment by Nishant2022** [comment]
We can merge this and I can create another PR.

**Review by korydraughn — APPROVED** [review]
Please add the pound and leave a message so @alanking knows to merge.

**Comment by Nishant2022** [comment]
@alanking Added # and ready to merge

**Review by alanking — APPROVED** [review]

**Pull request merged**